logwatch/logwatch_handlers.go

```diff
@@ -0,0 +1,173 @@
+package logwatch
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+	"github.com/smartcontractkit/wasp"
+)
+
+type LogTarget string
+
+const (
+	Loki LogTarget = "loki"
+	File LogTarget = "file"
+)
+
+type HandleLogTarget interface {
+	Handle(*ContainerLogConsumer, LogContent) error
+	PrintLogLocation(*LogWatch)
+}
+
+func getDefaultLogHandlers() map[LogTarget]HandleLogTarget {
+	handlers := make(map[LogTarget]HandleLogTarget)
+	handlers[Loki] = LokiLogHandler{
+		shouldSkipLogging: make(map[string]bool),
+	}
+	handlers[File] = FileLogHandler{
+		testLogFolders:    make(map[string]string),
+		shouldSkipLogging: make(map[string]bool),
+	}
+
+	return handlers
+}
+
+// streams logs to local files
+type FileLogHandler struct {
+	testLogFolders    map[string]string
+	shouldSkipLogging map[string]bool
+}
+
+func (h FileLogHandler) Handle(c *ContainerLogConsumer, content LogContent) error {
+	if val, ok := h.shouldSkipLogging[content.TestName]; val && ok {
```

Comment:
@Tofel maybe we could get rid of `shouldSkipLogging` field by removing the handler from `logTargetHandlers` in logwatch.go when there was an error? I think this would simplify the code in this file. WDYT?

Reply:
it would indeed! I tried to be smart here, so that if folder creation for one test fails the rest will try to create their folders anyway, but what are the chances of that actually happening? I guess very low.

Reply:
yeah, probably very low. Also, have you considered having separate `FileLogHandler` for each test?

Reply:
> EDIT: on the other hang with streaming logs to Loki it's probable that connection is intermitently lost and then re-established. If we remove the handler we will lose all logs since the time of first failure. If we keep a map per test we might still get logs from other tests.

Yeah, that makes sense!